docs/source/advanced/acls.markdown

```diff
@@ -0,0 +1,214 @@
+    resource_id = kwargs.get('resource_id')
+    # Some ACL checks also pass extra args in via kwargs, which we would
+    # access via:
+    #    resource_kwargs = kwargs.get('kwargs')
+    if authnz.user_is_user_type('user'):
+        return True
+    elif authnz.user_is_user_type('service'):
+        if resource_type == 'service' and action in ['metadata', 'get']:
+            # Does the resource ID match the authenticated username?
+            if authnz.user_is_service(resource_id):
+                return True
+        # We currently only allow services to access service get/metadata
+        return False
+    else:
+        # This should never happen, but paranoia wins out
+        return False
+```
+
+This function is defined by the `ACL_MODULE` setting, which by default is `confidant.authnz.rbac:default_acl`. The format is `python.path.to.module:function_in_module`. You can use this to implement an ACL approach that integrates with your own enviroment, or adjusts confidant's behavior to your needs.
+
+When implementing a new ACL module, remember that there's two types of users: `user` and `service`. By default `service` users are through kmsauth, whereas `user` users can come through kmsauth, or through one of the other forms of auth intended for the UI. You need to implement ACLs for both user types. Additionally, you should likely default your return to False, unless you're intending to only restrict features.
+
+## ACL Hookpoints
+
+The following hookpoints are currently available:
+
+### Credentials
+
+#### List credentials
+
+```python
+acl_module_check(resource_type='credential', action='list')
+```
+
+This check controls access to lists of credential metadata.
+
+#### Get credential metadata
+
+```python
+acl_module_check(
+    resource_type='credential',
+    action='metadata',
+    resource_id=id,
+)
+```
+
+This check controls access to specific credential metadata, which does not include credential pairs. Fine-grained controls can be applied using the provided `resource_id`.
+
+#### Get credential
+
+```python
+acl_module_check(
+    resource_type='credential',
+    action='get',
+    resource_id=id,
+)
+```
+
+This check controls access to specific credentials, which includes credential pairs. Fine-grained controls can be applied using the provided `resource_id`.
+
+#### Create credential
+
+```python
+acl_module_check(
+    resource_type='credential',
+    action='create',
+)
+```
+
+This check controls create access to credentials. This is a global permission, so no fine-grained ID is provided.
+
+#### Update credential
+
+```python
+acl_module_check(
+    resource_type='credential',
+    action='update',
+    resource_id=id,
+)
+```
+
+This check controls update access to specific credentials. Fine-grained controls can be applied using the provided `resource_id`. Note that if you're controlling access to this, you probably also want to control access to [Revert credential](#Revert credential).
```

Comment:
I think you may need to convert the anchor link from `#Revert credential` to `#revert-credential`